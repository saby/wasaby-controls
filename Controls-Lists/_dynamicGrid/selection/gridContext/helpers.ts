import { TColumnKey, TGridRowSelection, TGridSelection, TPlainSelection } from '../interfaces';

export function resetHasSiblingToDirection(
    selection: TGridRowSelection,
    direction: 'up' | 'down'
): boolean {
    let isChanged = false;

    Object.keys(selection.hasSibling).forEach((key) => {
        if (selection.hasSibling[key][direction]) {
            selection.hasSibling[key][direction] = false;
            isChanged = true;
        }
    });

    return isChanged;
}

function updateSiblingRowSelectionsToDirectionMutable(
    selection: TGridRowSelection,
    siblingSelection: TGridRowSelection,
    currentSelectionKey: TColumnKey,
    direction: 'up' | 'down'
): boolean {
    let isChanged = false;

    const siblingKey = Object.keys(siblingSelection.rowSelection).find(
        (prevKey) =>
            siblingSelection.rowSelection[prevKey].firstColumnKey ===
                selection.rowSelection[currentSelectionKey].firstColumnKey &&
            siblingSelection.rowSelection[prevKey].lastColumnKey ===
                selection.rowSelection[currentSelectionKey].lastColumnKey
    );

    if (typeof siblingKey !== 'undefined') {
        siblingSelection.hasSibling[siblingKey][direction === 'up' ? 'down' : 'up'] = true;
        selection.hasSibling[currentSelectionKey][direction] = true;
        isChanged = true;
    }

    return isChanged;
}

export function updateHasSiblingMutable(
    selection: TGridRowSelection,
    prevSiblingSelection: TGridRowSelection,
    nextSiblingSelection: TGridRowSelection
): [isPrevChanged: boolean, isNextChanged: boolean] {
    Object.keys(selection.rowSelection).forEach((key) => {
        selection.hasSibling[key] = {
            up: false,
            down: false,
        };
    });

    if (!prevSiblingSelection && !nextSiblingSelection) {
        return [false, false];
    }

    let isPrevChanged = false;
    let isNextChanged = false;

    if (prevSiblingSelection) {
        isPrevChanged = resetHasSiblingToDirection(prevSiblingSelection, 'down');
    }

    if (nextSiblingSelection) {
        isNextChanged = resetHasSiblingToDirection(nextSiblingSelection, 'up');
    }

    Object.keys(selection.rowSelection).forEach((currentSelectionKey) => {
        if (prevSiblingSelection) {
            isPrevChanged =
                updateSiblingRowSelectionsToDirectionMutable(
                    selection,
                    prevSiblingSelection,
                    currentSelectionKey as unknown as TColumnKey,
                    'up'
                ) || isPrevChanged;
        }

        if (nextSiblingSelection) {
            isNextChanged =
                updateSiblingRowSelectionsToDirectionMutable(
                    selection,
                    nextSiblingSelection,
                    currentSelectionKey as unknown as TColumnKey,
                    'down'
                ) || isNextChanged;
        }
    });

    if (isPrevChanged) {
        prevSiblingSelection.hasSibling = { ...prevSiblingSelection.hasSibling };
    }

    if (isNextChanged) {
        nextSiblingSelection.hasSibling = { ...nextSiblingSelection.hasSibling };
    }

    return [isPrevChanged, isNextChanged];
}

export function isEqualPlainSelections(
    selection1: TPlainSelection,
    selection2: TPlainSelection
): boolean {
    return compareKeys(Object.keys(selection1), Object.keys(selection2), (a, b) => {
        return compareKeys(selection1[a], selection2[b]);
    });
}

function compareKeys<T>(
    keys1: T[],
    keys2: T[],
    extraComparator?: (key1: T, key2: T) => boolean
): boolean {
    if (keys1.length !== keys2.length) {
        return false;
    } else if (keys1.length === 0) {
        return true;
    } else {
        const sorted1 = keys1.sort();
        const sorted2 = keys2.sort();

        // Ключи есть, их одинаковое количество и они отсортированы.
        for (let i = 0; i < sorted1.length; i++) {
            if (sorted1[i] !== sorted2[i]) {
                return false;
            } else if (extraComparator && !extraComparator(sorted1[i], sorted2[i])) {
                return false;
            }
        }

        return true;
    }
}

export function convertGridSelectionToPlainRecord(selection: TGridSelection): TPlainSelection {
    return Object.keys(selection).reduce((acc, sKey) => {
        return {
            ...acc,
            [sKey]: Object.keys(selection[sKey].rowSelection).reduce(
                (rowSelectedKeys, rsKey) => [
                    ...rowSelectedKeys,
                    ...selection[sKey].rowSelection[rsKey].selectedKeys,
                ],
                []
            ),
        };
    }, {});
}

export function convertPlainSelectionToGridSelection(selection: TPlainSelection): TGridSelection {
    const gridSelection: TGridSelection = {};
    // TODO: Не реализовано, будет делаться тут, если это вообще требуется.
    // Сейчас смена набора выбранных ячеек прикладным разработчиком через слайс не работает.
    return gridSelection;
}
