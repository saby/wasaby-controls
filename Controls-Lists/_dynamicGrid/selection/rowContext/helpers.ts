import { TColumnKey, ISelection, TRowSelection, TColumns } from '../interfaces';
import { datesEqualByQuantum, TQuantumType } from '../../render/utils';

export function getColumnIndexByKey(
    columnKey: TColumnKey,
    columns: TColumns,
    quantum: TQuantumType
): number {
    const date = new Date(columnKey);
    return columns.findIndex((d) => datesEqualByQuantum(date, d, quantum));
}

export function createNewSelection(newKeys: TColumnKey[]): ISelection {
    return {
        rootKey: newKeys[0],
        firstColumnKey: newKeys[0],
        lastColumnKey: newKeys[newKeys.length - 1],
        selectedKeys: newKeys,
        hasBorderTop: true,
        hasBorderBottom: true,
        gridColumnStart: undefined,
        gridColumnEnd: undefined,
    };
}

export function unshiftFromSelectionMutable(
    rowSelection: TRowSelection,
    selection: ISelection,
    columnsToRoot: Record<TColumnKey, TColumnKey>
): ISelection {
    const columnKey = selection.firstColumnKey;
    const rightSelection: ISelection = createNewSelection(selection.selectedKeys.slice(1));

    rightSelection.selectedKeys.forEach((k) => {
        columnsToRoot[k] = rightSelection.rootKey;
    });

    delete rowSelection[columnsToRoot[columnKey]];
    delete columnsToRoot[columnKey];

    rowSelection[rightSelection.rootKey] = rightSelection;

    return rightSelection;
}

export function popFromSelectionMutable(
    rowSelection: TRowSelection,
    selection: ISelection,
    columnsToRoot: Record<TColumnKey, TColumnKey>
): ISelection {
    const columnKey = selection.lastColumnKey;
    const leftSelection: ISelection = createNewSelection(
        selection.selectedKeys.slice(0, selection.selectedKeys.length - 1)
    );

    leftSelection.selectedKeys.forEach((k) => {
        columnsToRoot[k] = leftSelection.rootKey;
    });

    delete rowSelection[columnsToRoot[columnKey]];
    delete columnsToRoot[columnKey];
    rowSelection[leftSelection.rootKey] = leftSelection;

    return leftSelection;
}

export function splitSelectionMutable(
    rowSelection: TRowSelection,
    selection: ISelection,
    columnsToRoot: Record<TColumnKey, TColumnKey>,
    columnKey: TColumnKey
): [leftSelection: ISelection, rightSelection: ISelection] {
    const leftSelection = createNewSelection(
        selection.selectedKeys.slice(0, selection.selectedKeys.indexOf(columnKey))
    );

    leftSelection.selectedKeys.forEach((k) => {
        columnsToRoot[k] = leftSelection.rootKey;
    });

    const rightSelection = createNewSelection(
        selection.selectedKeys.slice(selection.selectedKeys.indexOf(columnKey) + 1)
    );

    rightSelection.selectedKeys.forEach((k) => {
        columnsToRoot[k] = rightSelection.rootKey;
    });

    delete rowSelection[columnsToRoot[columnKey]];
    delete columnsToRoot[columnKey];

    rowSelection[leftSelection.rootKey] = leftSelection;
    rowSelection[rightSelection.rootKey] = rightSelection;

    return [leftSelection, rightSelection];
}

export function deleteSelectionMutable(
    rowSelection: TRowSelection,
    selection: ISelection,
    columnsToRoot: Record<TColumnKey, TColumnKey>
): void {
    const columnKey = selection.rootKey;
    delete rowSelection[columnsToRoot[columnKey]];
    delete columnsToRoot[columnKey];
}

export function handleAddToSelectionMutable(
    rowSelection: TRowSelection,
    columnsToRoot: Record<TColumnKey, TColumnKey>,
    columnKey: TColumnKey,
    prevColumnKey: TColumnKey,
    nextColumnKey: TColumnKey
): ISelection {
    let joinSelection: ISelection;

    if (columnsToRoot[prevColumnKey]) {
        joinSelection = rowSelection[columnsToRoot[prevColumnKey]];

        joinSelection.selectedKeys.push(columnKey);
        joinSelection.lastColumnKey = columnKey;
        columnsToRoot[columnKey] = columnsToRoot[prevColumnKey];
    }

    if (columnsToRoot[nextColumnKey]) {
        const nextSelection = rowSelection[columnsToRoot[nextColumnKey]];
        const nextSelectionRoot = columnsToRoot[nextColumnKey];

        if (joinSelection) {
            joinSelection.selectedKeys.push(...nextSelection.selectedKeys);
            joinSelection.lastColumnKey = nextSelection.lastColumnKey;
            nextSelection.selectedKeys.forEach((key) => {
                columnsToRoot[key] = joinSelection.rootKey;
            });
            delete rowSelection[nextSelectionRoot];
        } else {
            joinSelection = nextSelection;

            joinSelection.selectedKeys.unshift(columnKey);
            joinSelection.firstColumnKey = columnKey;
            columnsToRoot[columnKey] = columnsToRoot[nextColumnKey];
        }
    }

    if (!joinSelection) {
        joinSelection = createNewSelection([columnKey]);
        rowSelection[columnKey] = joinSelection;
        columnsToRoot[columnKey] = columnKey;
    }

    return joinSelection;
}

export function getGridColumnCSSStyle(selection: ISelection, columns: TColumns, quantum: TQuantumType): {
    gridColumnStart: number;
    gridColumnEnd: number;
} {
    let gridColumnStart = getColumnIndexByKey(selection.firstColumnKey, columns, quantum);
    let gridColumnEnd = getColumnIndexByKey(selection.lastColumnKey, columns, quantum);

    if (gridColumnStart === -1 || gridColumnEnd === -1) {
        if (gridColumnStart === -1 && gridColumnEnd === -1) {
            gridColumnStart = undefined;
            gridColumnEnd = undefined;
        } else if (gridColumnStart === -1) {
            gridColumnStart = 1;
        } else if (gridColumnEnd === -1) {
            gridColumnEnd = -1;
        }
    } else {
        gridColumnStart = gridColumnStart + 1;
        gridColumnEnd = gridColumnEnd + 1 + 1;
    }

    return {gridColumnStart, gridColumnEnd};
}
