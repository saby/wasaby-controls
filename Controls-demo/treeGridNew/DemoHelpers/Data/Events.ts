import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
