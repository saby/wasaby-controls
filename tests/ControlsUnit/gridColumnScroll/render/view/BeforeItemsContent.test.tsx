/**
 * @jest-environment jsdom
 */
jest.mock('Controls/_gridColumnScroll/render/view/BeforeItemsContent/NavigationRow', () => {
    return {
        __esModule: true,
        default: (props) => {
            return (
                <div data-qa="mocked_NavigationRow">
                    <div data-qa="mocked_NavigationRow_props">{JSON.stringify(props, null, 4)}</div>
                </div>
            );
        },
    };
});

jest.mock('Controls/_gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow', () => {
    return {
        __esModule: true,
        default: (props) => {
            return (
                <div data-qa="mocked_ContentObserverRow">
                    <div data-qa="mocked_ContentObserverRow">{JSON.stringify(props, null, 4)}</div>
                </div>
            );
        },
    };
});

import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';

import { renderAnyComponent } from '../../forTestsOnly/helpers';

import {
    default as BeforeItemsContent,
    IBeforeItemsContentProps,
} from 'Controls/_gridColumnScroll/render/view/BeforeItemsContent';

describe('ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    it(
        'Нужно всегда рендерить переданный в опциях beforeItemsContent. ' +
            'Горизонтальный скролл расширяет функционал, а не перебивает его.',
        () => {
            const { asFragment } = renderAnyComponent(
                <BeforeItemsContent
                    part={'scrollable'}
                    columnScrollViewMode={'scrollbar'}
                    columnScrollNavigationPosition={undefined}
                    hasMultiSelectColumn={false}
                    stickyColumnsCount={2}
                    hasGridStickyHeader={true}
                >
                    <div>Какой то базовый контент</div>
                </BeforeItemsContent>,
                {
                    container,
                    wrapper: (props) => {
                        return <div className="wrapper">{props.children}</div>;
                    },
                }
            );

            expect(asFragment()).toMatchSnapshot();
        }
    );

    describe('Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition.', () => {
        const staticProps: IBeforeItemsContentProps = {
            hasMultiSelectColumn: false,
            stickyColumnsCount: 2,
            hasGridStickyHeader: true,

            part: undefined,
            columnScrollViewMode: undefined,
            columnScrollNavigationPosition: undefined,
        };

        const parts: IBeforeItemsContentProps['part'][] = ['fixed', 'scrollable'];
        const columnScrollViewModes: IBeforeItemsContentProps['columnScrollViewMode'][] = [
            'arrows',
            'scrollbar',
            'unaccented',
        ];
        const columnScrollNavigationPositions: IBeforeItemsContentProps['columnScrollNavigationPosition'][] =
            ['custom', undefined];

        parts.forEach((part) => {
            columnScrollViewModes.forEach((columnScrollViewMode) => {
                columnScrollNavigationPositions.forEach((columnScrollNavigationPosition) => {
                    it(`part=${part}, columnScrollViewMode=${columnScrollViewMode}, columnScrollNavigationPosition=${columnScrollNavigationPosition}.`, () => {
                        const { asFragment } = renderAnyComponent(
                            <BeforeItemsContent
                                {...staticProps}
                                part={part}
                                columnScrollViewMode={columnScrollViewMode}
                                columnScrollNavigationPosition={columnScrollNavigationPosition}
                            />,
                            {
                                container,
                                wrapper: (props) => {
                                    return <div className="wrapper">{props.children}</div>;
                                },
                            }
                        );

                        expect(asFragment()).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
