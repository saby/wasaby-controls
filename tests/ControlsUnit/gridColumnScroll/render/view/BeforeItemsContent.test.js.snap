// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Нужно всегда рендерить переданный в опциях beforeItemsContent. Горизонтальный скролл расширяет функционал, а не перебивает его. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "hasResizer": false,
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "scrollable",
    "hasResizer": false,
    "columnScrollViewMode": "scrollbar",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
    <div>
      Какой то базовый контент
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=arrows, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=arrows, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=arrows, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "fixed",
    "columnScrollViewMode": "arrows",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=arrows, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "fixed",
    "columnScrollViewMode": "arrows",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "fixed",
    "columnScrollViewMode": "scrollbar",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "fixed",
    "columnScrollViewMode": "scrollbar",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=unaccented, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=unaccented, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=unaccented, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=fixed, columnScrollViewMode=unaccented, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=arrows, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=arrows, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=arrows, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "scrollable",
    "columnScrollViewMode": "arrows",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=arrows, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "scrollable",
    "columnScrollViewMode": "arrows",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "scrollable",
    "columnScrollViewMode": "scrollbar",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=scrollbar, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
    <div
      data-qa="mocked_NavigationRow"
    >
      <div
        data-qa="mocked_NavigationRow_props"
      >
        {
    "part": "scrollable",
    "columnScrollViewMode": "scrollbar",
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false,
    "hasGridStickyHeader": true
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=unaccented, columnScrollNavigationPosition=custom. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=unaccented, columnScrollNavigationPosition=custom. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=unaccented, columnScrollNavigationPosition=undefined. 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent Рендер при различных значениях опций part, columnScrollViewMode и columnScrollNavigationPosition. part=scrollable, columnScrollViewMode=unaccented, columnScrollNavigationPosition=undefined. 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-qa="mocked_ContentObserverRow"
    >
      <div
        data-qa="mocked_ContentObserverRow"
      >
        {
    "stickyColumnsCount": 2,
    "hasMultiSelectColumn": false
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;
